genome_preparation.nf

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env nextflow
+
+def helpMessage() {
+    log.info"""
+    Index the provided genome with bowtie and samtools
+    Usage:
+            --fasta                     Fasta file containing the reference genome.
+            --outdir                    Directory where the index will be deployed.
+    """
+}
+
+// Show help message
+if (params.help) exit 0, helpMessage()
+
+params.fasta  = "$baseDir/resources/genome.fa"
+params.outdir = "$baseDir/resources/"
+
+/*
+ * Index the provided reference genome using bowtie build
+ */
+process bowtieGenomeIndex {
+
+    publishDir params.outdir,
+        overwrite: false,
+        mode: "move",
+        saveAs: { filename -> file(params.fasta).getName() + filename.replaceFirst(/index/, "") }
```

Comment:
I wish we didn't have to rename files every time using a closure... Do you think we could accept file basename as a parameter and pass it to all steps explicitly? This would allow us to avoid doing any renames like this, and it will probably mean easier integration with other steps in the future

Reply:
Yes those closures are not ideal at all and it took me a while to figure out how this actually works. Your suggestion is sensible and I'll give it a try.

Reply:
It is much better with your suggestion. thank you.